proxy-node-shared/src/main/java/uk/gov/ida/notification/contracts/verifyserviceprovider/Attributes.java

```diff
@@ -53,40 +55,82 @@ public Attributes(
         this.addresses = addresses;
     }
 
-    public List<Attribute<String>> getFirstNames() {
-        return firstNames != null ? firstNames : new ArrayList<>();
+    @JsonIgnore
+    public AttributesList<String> getFirstNames() {
+        return new AttributesList(firstNames, "firstName");
     }
 
-    public List<Attribute<String>> getMiddleNames() {
-        return middleNames != null ? middleNames : new ArrayList<>();
+    @JsonIgnore
+    public AttributesList<String> getMiddleNames() {
+        return new AttributesList(middleNames, "middleName");
     }
 
-    public List<Attribute<String>> getSurnames() {
-        return surnames != null ? surnames : new ArrayList<>();
+    @JsonIgnore
+    public AttributesList<String> getSurnames() {
+        return new AttributesList(surnames, "surname");
     }
 
-    public List<Attribute<DateTime>> getDatesOfBirth() {
-        return datesOfBirth != null ? datesOfBirth : new ArrayList<>();
+    @JsonIgnore
+    public AttributesList<DateTime> getDatesOfBirth() {
+        return new AttributesList(datesOfBirth, "dateOfBirth");
     }
 
+    @JsonIgnore
     public Attribute<String> getGender() {
         return gender;
     }
 
-    public List<Attribute<Address>> getAddresses() {
-        return addresses;
-    }
-
-    public static String combineAttributeValues(List<Attribute<String>> attributes) {
-        return attributes.stream()
-                .filter(Objects::nonNull)
-                .map(Attribute::getValue)
-                .filter(s -> !s.isEmpty())
-                .collect(Collectors.joining(" "));
+    @JsonIgnore
+    public AttributesList<Address> getAddresses() {
+        return new AttributesList(addresses, "address");
     }
 
     // Prints date in EIDAS format YYYY-MM-dd
-    public static String getFormattedDate(DateTime date) {
+    public static String getDateInEidasFormat(DateTime date) {
         return ISODateTimeFormat.date().withChronology(ISOChronology.getInstanceUTC()).print(date);
     }
+
+    public class AttributesList<T> {
+
+        private final List<Attribute<T>> attributes;
+        private final String type;
+
+        AttributesList(List<Attribute<T>> attributes, String type) {
+            this.attributes = Optional.ofNullable(attributes).orElse(Collections.emptyList());
+            this.type = type;
+        }
+
+        public String getType() {
+            return type;
+        }
+
+        public List<Attribute<T>> getValidAttributes() {
+
+            var current = getAllAttributes().stream()
+                    .filter(Attribute::isCurrent)
+                    .collect(Collectors.toList());
+
+            var verifiedAndCurrent = current.stream()
+                    .filter(Attribute::isVerified)
+                    .collect(Collectors.toList());
+
+            if (verifiedAndCurrent.isEmpty()) {
+                ProxyNodeLogger.info("No verified and current attributes: " + createAttributesMessage());
```

Comment:
I'm not clear why we might log at INFO level unless we are particularly interested in this case? We don't, for example, much care if there are no middle names?

Reply:
This is just FYI for us so we can diagnose why we're seeing missing first name errors for NL. This is an "expected error" but just need more details when it happens.

Reply:
OK, maybe put it a TODO to say that we're going to remove it again then? (or move the logging to somehow be specific to the first name case that we're interested in?)

Are we keeping the ticket open until we've actually checked the logs for this, then?

Reply:
We want to keep it forever - and it's not specific to just the first name. Basically, we want to know when there are missing attribute values so we can investigate and handle that case better. This is fine as it is IMO

Reply:
I definitely disagree that we want to create INFO log entries for ever for expected cases like user not having middle names. That just seems wrong to me.

Reply:
We can always adjust the logging in the future if needed. For now, we want to know when it happens